raiden/network/proxies/token_network.py

```diff
@@ -809,96 +864,241 @@ def _set_total_deposit(
-        transaction_executed: bool,
-        block_identifier: BlockSpecification,
-    ) -> Tuple[ErrorType, str]:
-        error_type: ErrorType = RaidenUnrecoverableError
-        msg = ""
-        latest_deposit = self._detail_participant(
-            channel_identifier=channel_identifier,
-            detail_for=self.node_address,
-            partner=partner,
-            block_identifier=block_identifier,
-        ).deposit
-
-        allowance = token.allowance(
-            owner=self.node_address,
-            spender=Address(self.address),
-            block_identifier=block_identifier,
-        )
-        if allowance < amount_to_deposit:
-            msg = (
-                "The allowance is insufficient. Check concurrent deposits "
-                "for the same token network but different proxies."
+            safety_deprecation_switch = self.safety_deprecation_switch(
+                block_identifier=failed_at_blockhash
             )
-        elif token.balance_of(self.node_address, block_identifier) < amount_to_deposit:
-            msg = "The address doesnt have enough tokens"
-        elif transaction_executed and latest_deposit < total_deposit:
-            msg = "The tokens were not transferred"
-        else:
-            participant_details = self.detail_participants(
-                participant1=self.node_address,
-                participant2=partner,
-                block_identifier=block_identifier,
-                channel_identifier=channel_identifier,
+            if safety_deprecation_switch:
+                msg = "This token_network has been deprecated."
+                raise RaidenRecoverableError(msg)
+
+            allowance = self.token.allowance(
+                owner=self.node_address,
+                spender=Address(self.address),
+                block_identifier=failed_at_blockhash,
             )
-            channel_state = self._get_channel_state(
-                participant1=self.node_address,
-                participant2=partner,
-                block_identifier=block_identifier,
-                channel_identifier=channel_identifier,
+            has_sufficient_balance = (
+                self.token.balance_of(self.node_address, given_block_identifier)
+                < amount_to_deposit
             )
-            # Check if deposit is being made on a nonexistent channel
-            if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            if allowance < amount_to_deposit:
                 msg = (
-                    f"Channel between participant {to_checksum_address(self.node_address)} "
-                    f"and {to_checksum_address(partner)} does not exist"
+                    "The allowance is insufficient. Check concurrent deposits "
+                    "for the same token network but different proxies."
+                )
+            elif has_sufficient_balance:
+                msg = "The address doesnt have enough tokens"
```

Comment:
This is never raising

Reply:
Could you please change it to be just:

```python
if condition_failed:
  raise
```
The pattern:
```python
if condition1_failed:
  msg = 'msg1'
elif condition2_failed:
  msg = 'msg1'
elif condition3_failed:
  msg = 'msg1'
raise Exception(msg0
```
Is quite error prone